src/Services/SchemaPoster.php

```diff
@@ -23,369 +23,382 @@
-    if (isset($_SERVER['ENV']) && $_SERVER['ENV'] === 'local') {
-      $response = $this->uploadSchemaAsZip($server_id);
+    /**
+     * Class Constructor.
+     *
+     * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+     *   Container Interface.
+     */
+    public function __construct(
+        ContainerInterface $container
+    ) {
+        $this->container = $container;
+        $this->logger = $container->get('logger.factory')->get('PantheonSolr');
+        $this->client = $container->get('search_api_pantheon.pantheon_guzzle');
     }
-    if (!$response instanceof Response) {
-      throw new \Exception('Cannot post schema to environment url.');
-    }
-    $log_function = in_array($response->getStatusCode(), [
-      200,
-      201,
-      202,
-      203,
-      204,
-    ])
-      ? 'notice'
-      : 'error';
-    $this->logger->{$log_function}('Files uploaded: {status_code} {reason}', [
-      'status_code' => $response->getStatusCode(),
-      'reason' => $response->getReasonPhrase(),
-    ]);
-    $message = vsprintf('Result: %s Status code: %d - %s', [
-      $log_function == 'error' ? 'NOT UPLOADED' : 'UPLOADED',
-      $response->getStatusCode(),
-      $response->getReasonPhrase(),
-    ]);
-    return [$message];
-  }
 
-  /**
-   * View a schema file on the pantheon solr server.
-   *
-   * @param string $filename
-   *   The filename to view. Default is Schema.xml.
-   *
-   * @return string|null
-   *   The text of the file or null on error or if the file doesn't exist.
-   */
-  public function viewSchema(string $filename = 'schema.xml'): ?string {
-    try {
-      $uri = (new Uri(
-        $this->getClient()
-          ->getEndpoint()
-          ->getCoreBaseUri() . 'admin/file'
-      ))->withQuery(
-        http_build_query([
-          'action' => 'VIEW',
-          'file' => $filename,
+    /**
+     * Post a schema file to the Pantheon Solr server.
+     *
+     * @param string $server_id
+     *   Search Api Server ID.
+     *
+     * @return array
+     *   Array of response messages.
+     *
+     * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+     * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+     * @throws \Drupal\search_api\SearchApiException
+     * @throws \Drupal\search_api_solr\SearchApiSolrException
+     */
+    public function postSchema(string $server_id): array
+    {
+        // PANTHEON Environment.
+        if (isset($_ENV['PANTHEON_ENVIRONMENT'])) {
+            $response = $this->uploadSchemaFiles($this->getSolrFiles($server_id));
+        }
+        // LOCAL DOCKER.
+        if (isset($_SERVER['ENV']) && $_SERVER['ENV'] === 'local') {
+            $response = $this->uploadSchemaAsZip($server_id);
+        }
+        if (!$response instanceof Response) {
+            throw new \Exception('Cannot post schema to environment url.');
+        }
+        $log_function = in_array($response->getStatusCode(), [
```

Comment:
Useless extra check, `guzzle` will throw an exception for anything non-2xx. Should be converted int `try/catch` block.